### PR TITLE
Syntax and Typing

### PR DESCRIPTION
**Pull request by mgbvox**
Added some syntax and typing annotations, and applied black/ruff pre-commit passes to code so pre-commit doesn't fail on machines where it is installed.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -109,15 +109,20 @@ autodoc_pydantic = { version = "*", optional = true }
...
+matplotlib = "^3.8.3"
 
 
 [tool.poetry.group.dev.dependencies]
+black = "^24.2.0"
+mypy = "^1.8.0"
+pre-commit = "^3.6.2"
 pytest = "^6.2.5"
-transformers = "^4.38.2"
-torch = "^2.2.1"
```
Comment (thomasahle) [inline pyproject.toml:117]: Do you need to change this?

File: pyproject.toml [hunk 2]
```diff
@@ -265,6 +270,14 @@ ignore = [
...
     # Sometimes we need List and Tuple
     "UP006",
+    # Asserts sometimes needed (e.g. assert X is not None)
+    "S101",
+    # Prints are currently being used for logging
+    "T201",
+    # .compile shadows a python builtin, and this is fine
+    "A003",
+    # Commented out code allowable for the time being
+    "ERA001",
```
Comment (thomasahle) [inline pyproject.toml:280]: I don't think we should have comment out code in the repo.

File: poetry.lock
Comment (thomasahle) [inline poetry.lock:1]: Do we need to change this?

**Review by thomasahle — COMMENTED** [review]
Remove the poetry lock file, unless it's really needed?
Maybe make sure the versioning stuff in pyproject is correct.
Other that that this looks good!

**Comment by okhat** [comment]
Will this affect other optimizers, like MIPRO?

**Pull request closed without merging**